swift-ticketing/test/swift_ticketing/handlers_test.clj

```diff
@@ -111,41 +81,184 @@
+                      expected-booking-id)]
+        (let [{:keys [status response]}
+              (client/reserve-ticket event-id reserve-ticket-req)
+              booking-id (get response "booking_id")]
+          (is (= status 201))
+          (is (= expected-booking-id booking-id)))))
+
+    (testing "with missing keys in request body"
+      (doseq [key (keys reserve-ticket-req)]
+        (let [reserve-ticket-req* (dissoc reserve-ticket-req key)
+              {:keys [status]}
+              (client/reserve-ticket event-id reserve-ticket-req*)]
+          (is (= status 400)
+              (str "Request without '" key "' should return 400")))))))
+
+(deftest reserve-ticket-test
+  (let [event-id (java.util.UUID/randomUUID)]
+    (testing "Reserving ticket (General)"
+      (reserve-ticket-test* #(factory/mk-reserve-general-ticket-request
+                              (inc (rand-int 10))
+                              (java.util.UUID/randomUUID)))
+      (testing "Reserving ticket (Seated)"
+        (reserve-ticket-test* #(factory/mk-reserve-seated-ticket-request
+                                [(java.util.UUID/randomUUID)]))))))
+
+(deftest make-payment-test
+  (testing "Payment handler"
+    (let [booking-id (str (java.util.UUID/randomUUID))
+          make-payment-called-with-correct-args (atom false)]
+      (with-redefs [booking/make-payment
+                    (fn [_ bid]
+                      (when (= booking-id bid)
+                        (reset! make-payment-called-with-correct-args true)))]
+        (let [{:keys [status response]} (client/make-payment booking-id)]
+          (is (= status 200))
+          (is @make-payment-called-with-correct-args)
+          (is (= booking-id (get response "booking_id"))))))))
+
+(deftest cancel-booking-test
+  (testing "Cancel Booking handler"
+    (let [booking-id (str (java.util.UUID/randomUUID))
+          cancel-booking-called-with-correct-args (atom false)]
+      (with-redefs [booking/cancel-booking
+                    (fn [_ bid]
+                      (when (= booking-id bid)
+                        (reset! cancel-booking-called-with-correct-args true)))]
+        (let [{:keys [status response]} (client/cancel-booking booking-id)]
+          (is (= status 200))
+          (is @cancel-booking-called-with-correct-args)
+          (is (= booking-id (get response "booking_id"))))))))
+
+(deftest get-booking-status-test
+  (testing "Get Booking Status handler"
+    (let [{:keys [db-spec]} fixtures/test-env
+          booking-id (str (java.util.UUID/randomUUID))
+          expected-booking-status :booking-status
+          get-booking-status-called-with-correct-args (atom false)]
+      (with-redefs [booking/get-booking-status
+                    (fn [dbs bid]
+                      (when (and
+                             (= db-spec dbs)
+                             (= booking-id bid))
+                        (reset! get-booking-status-called-with-correct-args true))
```

Comment:
This seems like a convoluted thing to do. You're putting a condition if something was called the way you expected it to be, and then setting an atom, and then putting assertions on the atom.
Why not put assertion on the condition itself?